### ISSUE TITLE
The results are inconsistent with the paper?

### ISSUE THREAD

**Issue by sty945** [wyy-code/DESAlign#1]
Hello friends,

When I obtained the results of the FB-DB15K dataset according to the current code, the early_stop in the log was 905, and I found that the H @ 1 result was always around 0.53. In the DBP_zh_n dataset, the early_stop in the log is 905, and the H @ 1 structure is maintained at around 0.79. This result differs from the results in the paper. 
I would like to know if the decrease in results is caused by hyperparameters.

```
bash run_desalign_00. sh 0 FBDB15K norm 0.2 1.0
bash run_desalign_00. sh 1 DBP15K zh_en 0.3 1.0
```

**Comment by wyy-code** [comment]
Hi there,

Yes, hyperparameter settings can significantly impact the results. In my main.py file, I've set self.early_stop_init = 300. Additionally, please be mindful of the device being used. During our experiments, we observed that results on the A100 were better than on a CPU, and on the A6000, the results didn't always reach the best performance. Could you also provide more details? The results provided are quite different from ours, and we haven't encountered this issue before.

Best regards

**Comment by sty945** [comment]
[DBP15K_zh_en_0.3_V1.log](https://github.com/user-attachments/files/16937924/DBP15K_zh_en_0.3_V1.log)
[FBDB15K_0.2_V1.log](https://github.com/user-attachments/files/16937925/FBDB15K_0.2_V1.log)

Thank you very much for your reply. The server I am currently using is Tesla P40. I have not made any changes to the code. The above are the dataset logs of FB-DB15K and DBP_zh_en that I obtained directly by running the program.
Looking forward to your reply.

**Comment by wyy-code** [comment]
[train.log](https://github.com/user-attachments/files/16950324/train.log)
I reran the code on an A800 GPU, and I’ve attached the log. I didn’t modify any code, but I noticed that my convergence speed was significantly slower than yours, while my accuracy was much higher. From my experience, the convergence speed on the A100 is similar to your results, usually completing in around 200 epochs.

I think there are three possible reasons for the differences:

1. cuDNN library might apply different optimizations depending on the GPU, and these operations are often non-deterministic by default. Even with the same random seed, this can lead to slightly different results on different devices. You can refer to this https://pytorch.org/docs/stable/notes/randomness.html#avoiding-nondeterministic-algorithms

2. Different GPUs support different precision levels. For example, the A100 trains in FP32, while older models might use mixed precision (FP16 + FP32) or other types, like TP32. https://www.nvidia.cn/data-center/products/a10-gpu/

3. There might be subtle differences in packages. In this test, I used CUDA 12.0 and PyTorch 1.13.1 on the A800, and the results also differed from my previous runs.

This suggests that the parameters in my file may not be optimal on these device, and you might need to manually adjust settings like the seed or batch size.

I hope this helps.

Best regards,
